pkg/controllers/managementuser/nodesyncer/nodessyncer.go

```diff
@@ -135,6 +135,10 @@ func Register(ctx context.Context, cluster *config.UserContext, kubeConfigGetter
 }
 
 func (n *nodeSyncer) sync(key string, node *corev1.Node) (runtime.Object, error) {
+	if node == nil || node.DeletionTimestamp != nil {
+		return nil, nil
+	}
+
 	needUpdate, err := n.needUpdate(key, node)
```

Comment:
Since the `needUpdate` function returns true if the provided node is nil or has a `DeletionTimestamp`, this function will trigger a resync when `needUpdate` evaluates to true. 
With your change, that specific case will be missed.